test/E2ETests/E2EApps/E2EApp/E2EApp.csproj

```diff
@@ -1,26 +1,39 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
-  <PropertyGroup>
+  <PropertyGroup Condition="$(DotnetVersion) != 'netfx'">
     <TargetFramework>net7.0</TargetFramework>
```

Comment:
Question: Can the `TargetFramework` value be a variable passed in from the caller of the test? We are officially supporting .net8 for the worker now. Wonder how easy it is to test on .net8 as well.

Reply:
For this PR, it was easy to run Net7 because everything else was already upgraded to Net7 as part of this old commit - https://github.com/Azure/azure-functions-dotnet-worker/pull/1203. We can certainly pass any variable from caller of the test but I believe we need to upgrade all the dependencies to be compatible with Net8 first.